contracts/QuestFactory.sol

```diff
@@ -249,7 +249,7 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
 
         quests[questId_].addressMinted[msg.sender] = true;
         quests[questId_].numberMinted++;
-        emit ReceiptMinted(msg.sender, questId_);
         rabbitHoleReceiptContract.mint(msg.sender, questId_);
+        emit ReceiptMinted(msg.sender, quests[questId_].questAddress, rabbitHoleReceiptContract.getTokenId(), questId_);
```

Comment:
Seeing the reentrancy attack from Slither here. I'll move the event before `mint()`, but will need to get token count and increase by 1 to get the correct tokenId

Reply:
I think it's fine to have the event emit after contract call, because we are not setting any storage in the emit.